session.cpp

```diff
@@ -139,11 +148,11 @@ shared_ptr<Device> Session::probe_device(libusb_device* device) {
 	}
 
 	if (desc.idVendor == 0x59e3 && desc.idProduct == 0xCEE1) {
-		dev = shared_ptr<Device>(new CEE_Device(this, device));
+		dev = (Device*)(new CEE_Device(this, device));
 	} else if (desc.idVendor == 0x0456 && desc.idProduct == 0xCEE2) {
-		dev = shared_ptr<Device>(new M1000_Device(this, device));
+		dev = (Device*)(new M1000_Device(this, device));
 	} else if (desc.idVendor == 0x064B && desc.idProduct == 0x784C) {
-		dev = shared_ptr<Device>(new M1000_Device(this, device));
+		dev = (Device*)(new M1000_Device(this, device));
```

Comment:
under what conditions is a shared pointer freed? the RC was presumably erroneously hitting zero and destroying a device prematurely.